### PR TITLE
Use more appropriate IntEnum and IntFlag types for some enumerations

### PR DESCRIPTION
**Pull request by rchl**
For types like `WatchKind` which is a bitfield kind, it makes sense to use `IntField` type so that bitwise operations are allowed on it.

For types that have numbers as values it makes sense to use `IntEnum` types so that it's allowed to compare with numbers.

### CODE REVIEW

**Comment by rchl** [comment]
This depends on the other PR so will need rebase after merging the other one.

**Comment by predragnikolic** [comment]
I think that this PR is ready for merging.

File: generate.py [hunk 1]
```diff
@@ -8,6 +8,9 @@
 from utils.helpers import get_new_literal_structures, reset_new_literal_structures, StructureKind
 
 
+BITWISE_ENUMS = ['WatchKind']
```
Comment (predragnikolic) [inline generate.py:11]: maybe move this to generate_enumerations file?

Comment (predragnikolic) [reply generate.py:11]: nah, it is ok here.

Comment (rchl) [reply generate.py:11]: I wasn't sure myself how generic the generation code is meant to be (whether it should have knowledge of specific LSP types)

**Pull request merged**